### PR TITLE
TSWeb-1274-Front-End-Development

### PR DESCRIPTION
**Pull request by VeronikaDolbina**
Added content and action tests for the Front-End Development page.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "packages": {
     "": {
       "devDependencies": {
-        "husky": "^8.0.3"
+        "husky": "^8.0.0"
```
Comment (Valerii-Kryvokhyzha) [inline package-lock.json:8]: why do we need to reduce version?

File: ui_tests_playwright/automation/preconditionsData/links/ClutchReviewLinks.ts [hunk 2]
```diff
@@ -6,4 +6,5 @@ export enum ClutchReviewLinks {
 	DerickDaily = 'https://clutch.co/profile/techstack#review-1634090',
 	AnonymousMedicalDevice = 'https://clutch.co/profile/techstack#review-1939515',
 	HenriYoki = 'https://clutch.co/profile/techstack#review-2196311',
+	Anonymous = 'https://clutch.co/profile/techstack#review-2232982',
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/preconditionsData/links/ClutchReviewLinks.ts:9]: could you pls add more context?
eg AnonymousFounderNjorda or AnonymousNjorda or smt else

File: ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentContent.test.ts [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+		'Granular reload',
+		'Extensive analytics',
+		'Integration for verification\nof customers',
+		'Improvements for\ncustomer engagement',
+	];
+
+	await expect(actualSectionTitles).toHaveText(expectedData);
+});
+
+test('Check carousel section numbers and titles in "Front-End Development Process" container from the "Front End Development" page @Regression @FrontEndDevelopment @TSWEB - 1274', async () => {
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentContent.test.ts:212]: @TSWEB-1274

File: ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentContent.test.ts [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+	await expect(carousel.getByTestId(Container.SectionTitle)).toHaveText([
+		'Discovery Phase',
+		'Pre-Engagement Phase',
+		'Engagement\nPhase',
+		'Delivery\nPhase',
+		'Release, maintenance\n& support',
+	]);
+});
+
+test('Check section titles in "Related Services" container from the "Front End Development" page @Regression @FrontEndDevelopment @TSWEB - 1274', async () => {
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentContent.test.ts:231]: @TSWEB-1274

File: ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentContent.test.ts [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+		'Consulting Services',
+		'Custom Software\nDevelopment',
+		'Big Data & Analytics',
+		'Internet of Things',
+	];
+
+	await expect(actualSectionTitles).toHaveText(expectedData);
+});
+
+test('Check section titles in "FAQ" container from the "Front End Development" page @Regression @FrontEndDevelopment @TSWEB - 1274', async () => {
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentContent.test.ts:248]: @TSWEB-1274

File: ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentContent.test.ts [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+	const expectedData = ['Team augmentation', 'Dedicated development team', 'R&D partnership'];
+
+	await expect(actualSectionTitles).toHaveText(expectedData);
+
+	await expect(cooperationModelsContainer.getByTestId(MainSiteButtons.ContactUsToDiscuss)).toHaveText(
+		'Contact us to discuss'
+	);
+});
+
+test('Check section titles in "How We Can Help" container from the "Front End Development" page @Regression @FrontEndDevelopment @TSWEB-1274', async () => {
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentContent.test.ts:180]: mb Check section numbers and titles...

File: ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentActions.test.ts [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+			.getByTestId(FrontEndDevelopment.FrontedWebDevelopment)
+			.getByTestId(MainSiteButtons.SubmitYourServiceRequestNow),
+		driver.getByTestId(FrontEndDevelopment.CooperationModels).getByTestId(MainSiteButtons.ContactUsToDiscuss),
+	];
+
+	for (const button of ctaButtons) {
+		await baseDriverSteps.checkScrollToContainerByCtaButtonClick(button, FrontEndDevelopment.GetInTouch);
+	}
+});
+
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentActions.test.ts:33]: could you pls add checks for redirecting by links in "Frontend Web Development Services" container?

File: ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentActions.test.ts [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+test('Check redirect by "Clutch Review" button in "Front-End Development Services Case Studies" container from the "Front End Development" page @Regression @FrontEndDevelopment @TSWEB-1274', async () => {
+	const frontEndCaseStudiesContainer = driver.getByTestId(FrontEndDevelopment.CaseStudy);
+	const clutchReviewButton = frontEndCaseStudiesContainer.getByTestId(Buttons.Clutch);
+
+	await baseDriverSteps.checkRedirectToPage(clutchReviewButton, ClutchReviewLinks.MarkBeare);
+});
+
+test('Check redirect by CTA button in "Case Studies" container from the "Front End Development" page @Regression @FrontEndDevelopment @TSWEB-1274', async () => {
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentActions.test.ts:41]: could you pls provide full container name?
Check redirect by CTA button in "Front-End Development Services Case Studies" container ....

File: ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentActions.test.ts [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+
+	await caseStudiesContainer.getByTestId(MainSiteButtons.ReadTheFullCaseStudy).click();
+	await baseDriverSteps.checkUrl(
+		UrlProvider.urlBuilder(
+			`${UrlPath.CaseStudies}${CaseStudyPath.DesignSystemForVideoStreamingPlatform}`,
+			Environment.Production
+		)
+	);
+});
+
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentActions.test.ts:52]: could you pls add check for redirect by "Clutch" link in "5.0-Star Clutch
Rating" section from "Why Techstack" container ?

**Review by Valerii-Kryvokhyzha — CHANGES_REQUESTED** [review]

File: ui_tests_playwright/automation/preconditionsData/links/ClutchReviewLinks.ts (hunk 2 above)
Comment (VeronikaDolbina) [reply ui_tests_playwright/automation/preconditionsData/links/ClutchReviewLinks.ts:9]: Sure. I fixed.

File: ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentActions.test.ts (hunk 7 above)
Comment (VeronikaDolbina) [reply ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentActions.test.ts:33]: I added a check.

File: ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentActions.test.ts [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+		[linkedInButtons.nth(0), ExpertsLinkedInLinks.VladyslavUshakov],
+		[linkedInButtons.nth(1), ExpertsLinkedInLinks.DmytroBohdanov],
+	]);
+
+	for (const [button, url] of buttonUrlMap) {
+		await baseDriverSteps.checkRedirectToPage(button, url);
+	}
+});
+
+test('Check redirects by Blog buttons in "Our Front End Development Experts" container from the "Front End Development" page @Regression @FrontEndDevelopment @TSWEB-1274', async () => {
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentActions.test.ts:105]: could you pls add @TSWEB-1061 tag to this test, to skip this test till the Blog will be Stable?

**Review by Valerii-Kryvokhyzha — CHANGES_REQUESTED** [review]

**Review by Valerii-Kryvokhyzha — CHANGES_REQUESTED** [review]

File: ui_tests_playwright/automation/test/ui/mainSite/services/backEndDevelopment/BackEndDevelopmentActions.test.ts [hunk 11]
```diff
@@ -127,7 +128,7 @@ test(
 		'Check redirect by "Clutch Review" button in "Why Techstack Stands Out" container from the "Back-End Development" page @Regression @BackEndDevelopment @TSWEB-1208'
 	),
 	async () => {
-		const whyTechstackContainer = driver.getByTestId(BackEndServices.WhyTechstack);
+		const whyTechstackContainer = driver.getByTestId(GeneralContainersMainSite.WhyTechstack);
```
[the remark this answers is not in the document]
Comment (Valerii-Kryvokhyzha) [reply ui_tests_playwright/automation/test/ui/mainSite/services/backEndDevelopment/BackEndDevelopmentActions.test.ts:128]: @VeronikaDolbina still actual

File: ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentActions.test.ts [hunk 12]
```diff
@@ -0,0 +1,154 @@
...
+import Buttons from '../../../../../identifiers/Buttons';
+import {ClutchReviewLinks} from '../../../../../preconditionsData/links/ClutchReviewLinks';
+import CaseStudyPath from '../../../../../providers/CaseStudyPath';
+import {Environment} from '../../../../../providers/EnvProvider';
+import {ExpertsLinkedInLinks} from '../../../../../preconditionsData/links/ExpertsLinkedInLinks';
+import {AuthorsEnum} from '../../../../../enum/AuthorsEnum';
+import Container from '../../../../../identifiers/Container';
+import MainSiteLinks from '../../../../../identifiers/mainSite/MainSiteLinks';
+import Links from '../../../../../preconditionsData/links/Links';
+import GeneralContainersMainSite from '../../../../../identifiers/mainSite/GeneralContainersMainSite';
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentActions.test.ts:17]: remove generalContainers pls and replace with appropriate page

File: ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentContent.test.ts [hunk 13]
```diff
@@ -0,0 +1,273 @@
...
+import {baseDriverSteps} from '../../../../../base/step/BaseDriverSteps';
+import UrlProvider from '../../../../../providers/UrlProvider';
+import UrlPath from '../../../../../providers/UrlPath';
+import {driver} from '../../../../../base/driver/Driver';
+import Container from '../../../../../identifiers/Container';
+import MainSiteButtons from '../../../../../identifiers/mainSite/MainSiteButtons';
+import FrontEndDevelopment from '../../../../../identifiers/mainSite/pages/services/FrontEndDevelopment';
+import MainSiteImages from '../../../../../identifiers/mainSite/MainSiteImages';
+import {ExpertNames} from '../../../../../preconditionsData/ExpertNames';
+import GeneralContainersMainSite from '../../../../../identifiers/mainSite/GeneralContainersMainSite';
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentContent.test.ts:11]: s

File: ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentContent.test.ts [hunk 14]
```diff
@@ -0,0 +1,273 @@
...
+		'Granular reload',
+		'Extensive analytics',
+		'Integration for verification\nof customers',
+		'Improvements for\ncustomer engagement',
+	];
+
+	await expect(actualSectionTitles).toHaveText(expectedData);
+});
+
+test('Check carousel section numbers and titles in "Front-End Development Process" container from the "Front End Development" page @desktop @mobile @Regression @FrontEndDevelopment @TSWEB-1274', async () => {
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentContent.test.ts:213]: pls add CTA

File: ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentActions.test.ts [hunk 15]
```diff
@@ -0,0 +1,154 @@
...
+	for (const [link, url] of linksUrlMap) {
+		await baseDriverSteps.checkRedirectToPage(link, url, UrlProvider.urlBuilder(UrlPath.FrontEndDevelopment));
+	}
+});
+
+test('Check redirect by "Clutch Review" button in "Front-End Development Services Case Studies" container from the "Front End Development" page @desktop @mobile @Regression @FrontEndDevelopment @TSWEB-1274', async () => {
+	const frontEndCaseStudiesContainer = driver.getByTestId(FrontEndDevelopment.CaseStudy);
+	const clutchReviewButton = frontEndCaseStudiesContainer.getByTestId(Buttons.Clutch);
+
+	await baseDriverSteps.checkRedirectToPage(clutchReviewButton, ClutchReviewLinks.MarkBeare);
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentActions.test.ts:61]: pls replace with checkRedirectToClutch()

File: ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentActions.test.ts [hunk 16]
```diff
@@ -0,0 +1,154 @@
...
+			Environment.Production
+		)
+	);
+});
+
+test('Check redirect by links in "Why Techstack" container from the "Front End Development" page @desktop @mobile @Regression @FrontEndDevelopment @TSWEB-1274', async () => {
+	const whyTechstackContainer = driver.getByTestId(GeneralContainersMainSite.WhyTechstack);
+	const clutchLink = whyTechstackContainer.getByTestId(MainSiteLinks.Clutch);
+
+	await baseDriverSteps.checkRedirectToPage(clutchLink, Links.ClutchReviews);
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentActions.test.ts:80]: pls replace with checkRedirectToClutch()

File: ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentActions.test.ts [hunk 17]
```diff
@@ -0,0 +1,154 @@
...
+});
+
+test('Check redirect by "Clutch Review" button in "Why Techstack" container from the "Front End Development" page @desktop @mobile @Regression @FrontEndDevelopment @TSWEB-1274', async () => {
+	const whyTechstackContainer = driver.getByTestId(GeneralContainersMainSite.WhyTechstack);
+
+	await baseDriverSteps.checkRedirectToPage(
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentActions.test.ts:86]: s

File: ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentActions.test.ts [hunk 18]
```diff
@@ -0,0 +1,154 @@
...
+		[blogButtons.nth(0), `${blogUri}${AuthorsEnum.VladyslavUshakov}`],
+		[blogButtons.nth(1), `${blogUri}${AuthorsEnum.DmytroBohdanov}`],
+	]);
+
+	for (const [button, url] of buttonUrlMap) {
+		await baseDriverSteps.checkRedirectToPage(button, url, UrlProvider.urlBuilder(UrlPath.FrontEndDevelopment));
+	}
+});
+
+test('Check carousel arrows clicks in "Front-End Development Process" container from the "Front End Development" page @desktop @mobile @Regression @FrontEndDevelopment @TSWEB-1274', async () => {
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentActions.test.ts:121]: remove @mobile tag, as on mobiles there is no arrows, only static arousel

File: ui_tests_playwright/automation/identifiers/mainSite/MainSiteButtons.ts [hunk 19]
```diff
@@ -12,6 +12,12 @@ export default class MainSiteButtons extends Buttons {
 	static GetYourFreeQuote = 'Button-GetYourFreeQuote';
 	static GetAFreeQuote = 'Button-GetAFreeQuote';
 	static CheckOutHowWeBuildIt = 'Button-CheckOutHowWeBuildIt';
+	static ReadTheFullCaseStudy = 'Button-ReadTheFullCaseStudy';
+	static ReadFullCaseStudy = 'Button-ReadFullCaseStudy';
+	static ReadFullCaseStudies = 'Button-ReadFullCaseStudies';
+	static AboutUs = 'Button-AboutUs';
+	static BookAMeeting = 'Button-BookAMeeting';
+	static TalkToAnExpert = 'Button-TalkToAnExpert';
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/identifiers/mainSite/MainSiteButtons.ts:20]: remove pls

File: ui_tests_playwright/automation/identifiers/mainSite/MainSiteButtons.ts [hunk 20]
```diff
@@ -24,6 +30,11 @@ export default class MainSiteButtons extends Buttons {
 	static GetYourFreeEstimate = 'Button-GetYourFreeEstimate';
 	static ChooseThisModel = 'Button-ChooseThisModel';
 	static BookYourExpertConsultation = 'Button-BookYourExpertConsultation';
+	static ScheduleAFreeBackendConsultation = 'Button-ScheduleAFreeBackendConsultation';
+	static MoreDetails = 'Button-MoreDetails';
+	static BoostYourBackend = 'Button-BoostYourBackend';
+	static LetsDiscussOurPartnership = 'Button-LetsDiscussOurPartnership';
+	static ReceiveAFreeQuote = 'Button-ReceiveAFreeQuote';
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/identifiers/mainSite/MainSiteButtons.ts:37]: remove pls

**Review by Valerii-Kryvokhyzha — CHANGES_REQUESTED** [review]

File: ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentActions.test.ts [hunk 21]
```diff
@@ -0,0 +1,154 @@
...
+		[linkedInButtons.nth(0), ExpertsLinkedInLinks.VladyslavUshakov],
+		[linkedInButtons.nth(1), ExpertsLinkedInLinks.DmytroBohdanov],
+	]);
+
+	for (const [button, url] of buttonUrlMap) {
+		await baseDriverSteps.checkRedirectToPage(button, url);
+	}
+});
+
+test('Check redirects by Blog buttons in "Our Front End Development Experts" container from the "Front End Development" page @desktop @mobile @Regression @FrontEndDevelopment @TSWEB-1061', async () => {
```
Comment (Valerii-Kryvokhyzha) [inline ui_tests_playwright/automation/test/ui/mainSite/services/frontEndDevelopment/FrontEndDevelopmentActions.test.ts:106]: pls also add @TSWEB-1274

**Review by Valerii-Kryvokhyzha — CHANGES_REQUESTED** [review]

**Pull request merged**